### ISSUE TITLE
How to test the model?

### ISSUE THREAD

**Issue by rezazad68** [XifengGuo/CapsNet-Keras#47]
I don't understand how to load weight for evaluation model. in code, you are making 3 different model for training, evaluating and manipulate. Since in the training process, the evaluation model does not involve, I was wondering how it can be used for testing? we can't load training weight on evaluation model since they have different structures.

**Comment by XifengGuo** [comment]
@rezazad68 The training model, evaluation model and manipulation model will share the same weights since they share most of the computation graph. You can validate this by writing a simple demo.

**Comment by rezazad68** [comment]
Thank you very much, I already solved my problem. in the training section, you are giving test set as a validation set, is not it incorrect? I was wondering to know that, does this way is true for reporting accuracy?

**Comment by XifengGuo** [comment]
@rezazad68
The right way to report the performance is to split the whole dataset into training, validation, and testing set where the testing accuracy is the right one. As I don't use the validation accuracy to control the training process (like early stopping), It's alright take the final validation accuracy as testing accuracy. Of course, it'll be better to follow the standard splitting procedure.
